spyder/plugins/workingdirectory/container.py

```diff
@@ -78,16 +78,6 @@ def enterEvent(self, event):
         """Set current path as the tooltip of the widget on hover."""
         self.setToolTip(self.currentText())
 
-    def focusOutEvent(self, event):
```

Comment:
Unfortunately, removing this method has an unexpected side effect: if a not valid directory is left in the toolbar, then the toolbar becomes empty on focus out:

![cwd-blank](https://github.com/user-attachments/assets/d4fc4f76-f10b-4688-9341-848e31554af5)

On master we restore the last valid cwd:

![cwd-on-master](https://github.com/user-attachments/assets/45f790cd-b1e8-4ee7-8fb0-bc6b0b942ec6)

Perhaps a simpler solution could be to block Qt signals when calling `selected` to prevent `open_dir` to be emitted (but I'm not sure about it).

Reply:
Forgot to add a comment here but I think the last couple of commits should fix this + add some tests for the validation and fallback functionality on focus out

Reply:
Thanks for the update @dalthviz! Now the combobox is not left blank when an invalid directory is set on it, but the new directory is not automatically set by Spyder.

In the gif below you can see that the initial cwd is `/home/carlos/tmp`. Then I left `/home/carlos/tm` on the combobox, and on focus out the directory was set to `/home/carlos`, which is good. But no signal was emitted after that, so Files is still showing the contents of `/home/carlos/tmp`.  

![cwd-parent-not-set](https://github.com/user-attachments/assets/1ebb4b92-2eb0-4c33-99d1-66f75310d89b)

Reply:
Commit [87c4237](https://github.com/spyder-ide/spyder/pull/24620/commits/87c4237acc3039fbbf601a6177d6377cf0729c22) extended the added tests to check for an actual cwd change to pass and commit [a0243f9](https://github.com/spyder-ide/spyder/pull/24620/commits/a0243f9abdc6c56f646bc30de413c3d21a4e39f4) should fix the behavior above

Reply:
Checked locally and works as expected now, thanks @dalthviz!